--- a/PlSqlLexer.g4
+++ b/PlSqlLexer.g4
@@ -2594,11 +2594,11 @@
 MULTI_LINE_COMMENT  : '/*' .*? '*/'                    -> channel(HIDDEN);
 // https://docs.oracle.com/cd/E11882_01/server.112/e16604/ch_twelve034.htm#SQPUG054
 REMARK_COMMENT:
-    'REM' {this.IsNewlineAtPos(-4)}? 'ARK'? (' ' ~('\r' | '\n')*)? NEWLINE_EOF -> channel(HIDDEN)
+    'REM' {self.IsNewlineAtPos(-4)}? 'ARK'? (' ' ~('\r' | '\n')*)? NEWLINE_EOF -> channel(HIDDEN)
 ;
 
 // https://docs.oracle.com/cd/E11882_01/server.112/e16604/ch_twelve032.htm#SQPUG052
-PROMPT_MESSAGE: 'PRO' {this.IsNewlineAtPos(-4)}? 'MPT'? (' ' ~('\r' | '\n')*)? NEWLINE_EOF;
+PROMPT_MESSAGE: 'PRO' {self.IsNewlineAtPos(-4)}? 'MPT'? (' ' ~('\r' | '\n')*)? NEWLINE_EOF;
 
 REGULAR_ID: SIMPLE_LETTER (SIMPLE_LETTER | '$' | '_' | '#' | [0-9])*;
 
--- a/PlSqlParser.g4
+++ b/PlSqlParser.g4
@@ -2289,7 +2289,7 @@
 
 // https://docs.oracle.com/database/121/SQLRF/statements_4008.htm#SQLRF56110
 unified_auditing
-    : {this.isVersion12()}? AUDIT (
+    : {self.isVersion12()}? AUDIT (
         POLICY policy_name ((BY | EXCEPT) audit_user (',' audit_user)*)? (WHENEVER NOT? SUCCESSFUL)?
         | CONTEXT NAMESPACE oracle_namespace ATTRIBUTES attribute_name (',' attribute_name)* (
             BY audit_user (',' audit_user)*
@@ -2314,11 +2314,11 @@
     ;
 
 audit_direct_path
-    : {this.isVersion12()}? DIRECT_PATH auditing_by_clause
+    : {self.isVersion12()}? DIRECT_PATH auditing_by_clause
     ;
 
 audit_container_clause
-    : {this.isVersion12()}? (CONTAINER EQUALS_OP (CURRENT | ALL))
+    : {self.isVersion12()}? (CONTAINER EQUALS_OP (CURRENT | ALL))
     ;
 
 audit_operation_clause
@@ -2362,7 +2362,7 @@
         object_name
         | DIRECTORY regular_id
         | MINING MODEL model_name
-        | {this.isVersion12()}? SQL TRANSLATION PROFILE profile_name
+        | {self.isVersion12()}? SQL TRANSLATION PROFILE profile_name
         | DEFAULT
     )
     ;
@@ -2390,7 +2390,7 @@
     | MATERIALIZED VIEW
     | NOT EXISTS
     | OUTLINE
-    | {this.isVersion12()}? PLUGGABLE DATABASE
+    | {self.isVersion12()}? PLUGGABLE DATABASE
     | PROCEDURE
     | PROFILE
     | PUBLIC DATABASE LINK
@@ -2632,11 +2632,11 @@
     ;
 
 library_editionable
-    : {this.isVersion12()}? (EDITIONABLE | NONEDITIONABLE)
+    : {self.isVersion12()}? (EDITIONABLE | NONEDITIONABLE)
     ;
 
 library_debug
-    : {this.isVersion12()}? DEBUG
+    : {self.isVersion12()}? DEBUG
     ;
 
 compiler_parameters_clause
@@ -2714,7 +2714,7 @@
     ;
 
 alter_view_editionable
-    : {this.isVersion12()}? (EDITIONABLE | NONEDITIONABLE)
+    : {self.isVersion12()}? (EDITIONABLE | NONEDITIONABLE)
     ;
 
 // https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/CREATE-VIEW.html
@@ -4556,7 +4556,7 @@
     ;
 
 partial_database_recovery_10g
-    : {this.isVersion10()}? STANDBY (
+    : {self.isVersion10()}? STANDBY (
         TABLESPACE tablespace (',' tablespace)*
         | DATAFILE CHAR_STRING
         | filenumber (',' CHAR_STRING | filenumber)*
@@ -5479,7 +5479,7 @@
     ;
 
 period_definition
-    : {this.isVersion12()}? PERIOD FOR column_name ('(' start_time_column ',' end_time_column ')')?
+    : {self.isVersion12()}? PERIOD FOR column_name ('(' start_time_column ',' end_time_column ')')?
     ;
 
 start_time_column
@@ -6212,7 +6212,7 @@
     ;
 
 table_ref_aux
-    : table_ref_aux_internal flashback_query_clause* ({this.isNotStartOfJoin()}? table_alias)?
+    : table_ref_aux_internal flashback_query_clause* ({self.isNotStartOfJoin()}? table_alias)?
     ;
 
 table_ref_aux_internal
@@ -6557,7 +6557,7 @@
     | json_table_clause (AS identifier)?
     | LATERAL '(' subquery subquery_restriction_clause? ')'
     // Deprecated Oracle 10/11 RELATIONAL alias for casting object-types to relational tables
-    | {this.isVersion11()}? (RELATIONAL '(' tableview_name NOT XMLTYPE ')')
+    | {self.isVersion11()}? (RELATIONAL '(' tableview_name NOT XMLTYPE ')')
     ;
 
 table_collection_expression
@@ -6722,7 +6722,7 @@
     : case_expression
     | quantified_expression
     | standard_function
-    | {this.IsNotNumericFunction()}? atom
+    | {self.IsNotNumericFunction()}? atom
     | implicit_cursor_expression
     ;
 
@@ -6817,7 +6817,7 @@
     : string_function
     | numeric_function_wrapper
     | json_function
-    | {this.IsNotNumericFunction()}? other_function
+    | {self.IsNotNumericFunction()}? other_function
     ;
 
 //see as https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/JSON_ARRAY.html#GUID-46CDB3AF-5795-455B-85A8-764528CEC43B
